### ISSUE TITLE
Protelis 12.1 TestStacktrace failing

### ISSUE THREAD

**Issue by jakebeal** [Protelis/Protelis#230]
When running protelis-test for Protelis 12.1.0 as JUnit, the org.protelis.test.TrestStacktrace.testErrorTraceModule test fails for me.
This is with a fresh build in Eclipse Version: 2018-12 (4.10.0)

The error is as follows:

```
java.lang.AssertionError: Exception does not include argument type mismatch identification
org.protelis.lang.interpreter.util.ProtelisRuntimeException: null: cannot invoke public static double java.lang.Math.sin(double) with arguments (I'm definitely not a number.: String)
	at errorTrace02:rootError at line number 8
	at errorTrace02:aCall at line number 14
	at errorTrace02:anotherCall between lines 20 and 22
Fully detailed interpreter trace:
org.protelis.lang.interpreter.util.ProtelisRuntimeException: null: cannot invoke public static double java.lang.Math.sin(double) with arguments (I'm definitely not a number.: String)
	at: sin(I'm definitely not a number.) at line number 8
	at: nbr(sin) between lines 7 and 9
	at: minHood(nbr) between lines 6 and 10
	at: rep (x <- 1.0) { minHood } between lines 5 and 11
	at: errorTrace02:rootError(){ rep } at line number 14
	at: nbr(errorTrace02:rootError) at line number 14
	at: errorTrace02:aCall(){ nbr } at line number 22
	at: errorTrace02:anotherCall(){ +; errorTrace02:aCall } at line number 28
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.protelis.test.TestStacktrace.lambda$1(TestStacktrace.java:30)
	at org.protelis.test.ProgramTester.runExpectingErrors(ProgramTester.java:116)
	at org.protelis.test.TestStacktrace.testErrorTraceModule(TestStacktrace.java:28)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:305)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:365)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:330)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:78)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:328)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:65)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:292)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:305)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:412)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:89)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:41)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:541)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:763)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:463)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:209)
```

**Comment by DanySK** [comment]
It's a MacOS-specific issue due to inconsistent behavior of the JVM already intercepted by the new CI and solved in 129ff5005a75d633f15d9c9c22f431bb91bc3cf5.

**Comment by jakebeal** [comment]
If it's fixed, shouldn't this not be an issue in 12.1?

**Comment by DanySK** [comment]
It was fixed after 12.1.0 was released. If you checkout develop, the issue should be gone. I did not cut a 12.1.1 release. Do we want to??

**Comment by jakebeal** [comment]
I was confused because the commit you reference is 5 days older than 12.1.0.

**Comment by DanySK** [comment]
Now I am dubious as well: can you try if the issue stands in develop? I do not have access to a MacOS machine, and CI testing apparently works flawlessly

**Comment by jakebeal** [comment]
When I run `develop`, I do not see the failure.

**Comment by jakebeal** [comment]
Looks like this should be resolved by the 12.2 release, then.
